src/main/cpp/src/json_parser.hpp

```diff
@@ -0,0 +1,1786 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/string_view.hpp>
+#include <cudf/types.hpp>
+
+#include <thrust/pair.h>
+#include <thrust/tuple.h>
+
+namespace spark_rapids_jni {
+
+/**
+ * write style when writing out JSON string
+ */
```

Comment:
My comment for the docs in this file is similar to the previous one: Don't use non-standard oxygen. I'm fine to leave them for now but we have to cleanup in the follow up work.

Reply:
Will do in a follow-on.